### ISSUE TITLE
yubioath-desktop unable to import 'yubikey' Python module

### ISSUE THREAD

**Issue by aidalgol** [NixOS/nixpkgs#202295]
### Describe the bug
yubioath-desktop is unable to talk to YubiKey devices, seemingly because it is unable to import the `yubikey` python module.

### Steps To Reproduce
1. Launch `yubioath-desktop` from a terminal

```console
qrc:/qml/main.qml:305:5: QML Shortcut: Shortcut: Only binding to one of multiple key bindings associated with 7. Use 'sequences: [ <key> ]' to bind to all of them.
qrc:/qml/main.qml:297:5: QML Shortcut: Shortcut: Only binding to one of multiple key bindings associated with 9. Use 'sequences: [ <key> ]' to bind to all of them.
Qt Quick Layouts: Detected recursive rearrange. Aborting after two iterations.
Qt Quick Layouts: Detected recursive rearrange. Aborting after two iterations.
"PyOtherSide error: Traceback (most recent call last):\n\n  File \"qrc:///py/yubikey.py\", line 23, in <module>\n    from ykman.device import scan_devices, list_all_devices, connect_to_device, get_name, read_info\n\nImportError: cannot import name 'connect_to_device' from 'ykman.device' (/nix/store/r7ns0h8gd7f2rm27m2dwxxv9qzr22kqy-python3.10-yubikey-manager-5.0.0/lib/python3.10/site-packages/ykman/device.py)\n"
Unhandled PyOtherSide error: Cannot import module: yubikey (Traceback (most recent call last):

  File "qrc:///py/yubikey.py", line 23, in <module>
    from ykman.device import scan_devices, list_all_devices, connect_to_device, get_name, read_info

ImportError: cannot import name 'connect_to_device' from 'ykman.device' (/nix/store/r7ns0h8gd7f2rm27m2dwxxv9qzr22kqy-python3.10-yubikey-manager-5.0.0/lib/python3.10/site-packages/ykman/device.py)
)
"PyOtherSide error: Traceback (most recent call last):\n\n  File \"<string>\", line 1, in <module>\n\nNameError: name 'yubikey' is not defined\n"
Unhandled PyOtherSide error: Function not found: 'yubikey.init' (Traceback (most recent call last):

  File "<string>", line 1, in <module>

NameError: name 'yubikey' is not defined
)
"PyOtherSide error: Traceback (most recent call last):\n\n  File \"<string>\", line 1, in <module>\n\nNameError: name 'yubikey' is not defined\n"
Unhandled PyOtherSide error: Function not found: 'yubikey.controller.check_descriptors' (Traceback (most recent call last):

  File "<string>", line 1, in <module>

NameError: name 'yubikey' is not defined
)
qml: TypeError: Cannot read property 'success' of undefined undefined
"PyOtherSide error: Traceback (most recent call last):\n\n  File \"<string>\", line 1, in <module>\n\nNameError: name 'yubikey' is not defined\n"
Unhandled PyOtherSide error: Function not found: 'yubikey.controller.is_win_non_admin' (Traceback (most recent call last):

  File "<string>", line 1, in <module>

NameError: name 'yubikey' is not defined
)
qml: TypeError: Cannot read property 'winNonAdmin' of undefined undefined
```

### Expected behavior
No Python errors, and the TOTP tokens to show up in the UI.

### Screenshots
This is what the UI displays, indefinitely.
![image](https://user-images.githubusercontent.com/2313201/203219875-e441fbfe-828c-49db-9e67-a6e0e7ab788a.png)

### Additional context
The behaviour is the same regardless of whether a YubiKey device is present.

`yubikey-oath` is installed via `home-manager`, using a flake-based config.

### Notify maintainers
@Mic92 @rissson 

### Metadata
```console
❯ nix-shell -p nix-info --run "nix-info -m"
this path will be fetched (0.00 MiB download, 0.00 MiB unpacked):
  /nix/store/a2mfyh90szmc055ivfnv182b5a4naq90-nix-info
copying path '/nix/store/a2mfyh90szmc055ivfnv182b5a4naq90-nix-info' from 'https://cache.nixos.org'...
 - system: `"x86_64-linux"`
 - host os: `Linux 5.15.78, NixOS, 22.05 (Quokka), 22.05.20221114.16f4e04`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.8.1`
 - channels(aidan): `"nixpkgs"`
 - channels(root): `"nixos"`
 - nixpkgs: `/home/aidan/.nix-defexpr/channels/nixpkgs`
```

**Comment by Mic92** [comment]
It's not clear to me, which python package they want here.

**Comment by Mic92** [comment]
It looks like upstream has deprecated this package in favor of: https://github.com/Yubico/yubioath-flutter

**Comment by Mic92** [comment]
See https://github.com/NixOS/nixpkgs/pull/202411

**Comment by 0xf09f95b4** [comment]
Hi!

The error is caused by a dependency on yubikey-manager5, while yubioath-desktop still needs yubikey-manager4, because the python libs had breaking changes (see https://github.com/Yubico/yubikey-manager-qt/issues/328).

By dropping the `yubikey-manager` dependency to `yubikey-manager4`, this can be fixed.

A workaround:
```nix
nixpkgs.config.packageOverrides = pkgs: {
   yubioath-desktop = pkgs.yubioath-desktop.override { yubikey-manager = pkgs.yubikey-manager4; };
}; 
```

A suggested fix:

Change https://github.com/NixOS/nixpkgs/blob/nixos-22.11/pkgs/applications/misc/yubioath-desktop/default.nix:

```diff
, yubikey-personalization
-, yubikey-manager
+,yubikey-manager4
, makeWrapper
```

I know that this package is dropped in master (https://github.com/NixOS/nixpkgs/pull/202411) but this could be a relatively simple fix for 22.11, before changing to the new flutter-based gui.

Thanks!

**Comment by rkjnsn** [comment]
Thanks for the suggested fix, @0xf09f95b4. Submitted a PR: https://github.com/NixOS/nixpkgs/pull/209098. 🤞🏻 it gets picked up.
